uint8 phase
string description